### DISCUSSION TITLE
Use ControlNet Reference Only to change the clothes of a character, with one photo

### DISCUSSION THREAD

**Discussion by andupotorac** [Bing-su/adetailer#141] [Ideas]
So I used this vest photo:
![vest](https://github.com/Bing-su/adetailer/assets/3269738/013819a9-8957-4ab1-86ed-4cce11b7a511)

And these settings:
![Reference Only - same clothes](https://github.com/Bing-su/adetailer/assets/3269738/75151ead-5700-4e87-8131-d8e9360eb34c)

To dress any kind of LORAs with it: 
<img width="2093" alt="image" src="https://github.com/Bing-su/adetailer/assets/3269738/d4b1c87f-dcf5-4a53-99cb-032c61e2dbd5">

Would it be possible to solve the issue of clothes that a person is wearing using this? It doesn't seem to need any algorithms to detect faces or hands, can use something like BLIP to describe the garment in 2-3 words (eg. floral vest, black pants, yellow sweater, etc) so that all can be automatic.

So at the end you could:
- fix the face
- fix the hands
- dress the character in the clothes of your choice

**Comment by Rkkss** [comment]
think you still need seg dectetion model to detect clothes and inpaint them out.
something like this, where you can prompt the clothes and let model map it: https://huggingface.co/docs/transformers/main/en/model_doc/clipseg

**Comment by daffatahtaa** [comment]
Hello,

been trying to recreate what you did but why is my generated image doesn't resemble the reference at all?
![image](https://github.com/Bing-su/adetailer/assets/43267539/f04d5d8e-d751-4a10-909a-ca9d0b9a8cd7)
